src/main/java/org/jabref/gui/entryeditor/CommentsTab.java

```diff
@@ -139,17 +138,17 @@ protected void setupPanel(BibEntry entry, boolean compressed) {
         // Show "Hide" button only if user-specific comment field is empty. Otherwise, it is a strange UI, because the
         // button would just disappear and no change **in the current** editor would be made
         if (entryEditorPreferences.shouldShowUserCommentsFields() && !entry.hasField(userSpecificCommentField)) {
-            Button hideDefaultOwnerCommentButton = new Button(Localization.lang("Hide user comments"));
-            hideDefaultOwnerCommentButton.setOnAction(e -> {
-                var labelForField = gridPane.getChildren().stream().filter(s -> s instanceof FieldNameLabel).filter(x -> ((FieldNameLabel) x).getText().equals(userSpecificCommentField.getDisplayName())).findFirst();
-                labelForField.ifPresent(label -> gridPane.getChildren().remove(label));
-                fieldEditorForUserDefinedComment.ifPresent(f -> gridPane.getChildren().remove(f.getNode()));
-                editors.remove(userSpecificCommentField);
-
-                entryEditorPreferences.setShowUserCommentsFields(false);
-                setupPanel(entry, false);
+            Button toggleCommentsButton = new Button("Hide user comments");
+            toggleCommentsButton.setOnAction(e -> {
+                if (gridPane.getChildren().contains(fieldEditorForUserDefinedComment.get().getNode())) {
+                    gridPane.getChildren().remove(fieldEditorForUserDefinedComment.get().getNode());
```

Comment:
You don't have to remove the node from the layout, just `setVisible(false)` but if you decided on using a toggle button this won't be necessary.